contracts/multisig-prover/src/execute.rs

```diff
@@ -146,6 +158,7 @@ fn get_next_worker_set(
     deps: &DepsMut,
     env: &Env,
     config: &Config,
+    called_by_update_worker_set: bool,
```

Comment:
I would avoid something like this, for two reasons: 1. at the callsite, it's not clear what a random boolean flag is for. And 2. It increases the complexity of the function, by adding more conditional branches.

What if we refactor this so this just returns an error if there is no current worker set? And then make a separate function, like `make_worker_set`, that is just:
```
    let workers_info = get_workers_info(deps, env, config)?;
    return WorkerSet::new(
            workers_info.pubkeys_by_participant,
            workers_info.snapshot.quorum.into(),
            env.block.height,
        )?;

```